### PR TITLE
Feature/fetchcall refactor

### PR DESCRIPTION
**Pull request by Eleanorgruth**
Summary of the work done:

- Fetch calls, bug fix for duplicates

This branch includes:

- Ingredient class and ingredient tests

Contributed to code:

- Everyone!

Needs to review:

- Everyone!

### CODE REVIEW

File: src/apiCalls.js [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
+let getIngredientsData = fetch("https://what-s-cookin-starter-kit.herokuapp.com/api/v1/ingredients")
+    .then(response => response.json())
+    .then(data => {return data})
+    .catch(err => console.log(err))
+
+let getUserData = fetch("https://what-s-cookin-starter-kit.herokuapp.com/api/v1/users")
+    .then(response => response.json())
+    .then(data => {return data})
+    .catch(err => console.log(err))
+
```
Comment (FrazierLE) [inline src/apiCalls.js:17]: change let to const

File: src/classes/Recipe.js [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
-        this.tags = data.tags;
+    constructor(recipeData) {
+        this.id = recipeData.id;
+        this.image = recipeData.image;
+        this.ingredients = recipeData.ingredients;
+        this.instructions = recipeData.instructions;
+        this.name = recipeData.name;
+        this.tags = recipeData.tags;
         this.ingredientsList = [];
         this.totalCost;
```
Comment (FrazierLE) [inline src/classes/Recipe.js:13]: Are you passing in the sample data or the api data in our class?

File: src/scripts.js [hunk 3]
```diff
@@ -1,22 +1,44 @@
...
 import './styles.css';
 import RecipeRepository from './classes/RecipeRepository';
 import Recipe from "./classes/Recipe";
 import User from "./classes/User";
-import { ingredientsData } from './data/ingredients'
-import { recipeData } from './data/recipes';
-import { usersData } from './data/users';
+
 
 // An example of how you tell webpack to use an image (also need to link to it in the index.html)
```
Comment (knpausch) [inline src/scripts.js:8]: should we delete this later? do we need it?

File: src/scripts.js [hunk 4]
```diff
@@ -33,12 +55,13 @@ const saveRecipeButton = document.querySelector('#favorite-recipe-button')
...
+const deleteInstructions = document.querySelector('#delete-instructions')
 
 
 // ~~~~~~~~~~~~~~ Event Listeners ~~~~~~~~~~~~~~~~~~~~
 allRecipes.addEventListener('click', viewRecipeDetail);
-window.addEventListener('load', displayAllRecipes);
-window.addEventListener('load', displayWelcomeMessage);
+//window.addEventListener('load', displayAllRecipes);
+//window.addEventListener('load', displayWelcomeMessage);
 savedButton.addEventListener('click', displayFavorites);
```
Comment (FrazierLE) [inline src/scripts.js:65]: Let's take this out since we updated the window load with fetch

File: src/scripts.js [hunk 5]
```diff
@@ -1,22 +1,44 @@
...
 let foundRecipe;
+//let listOfIngredients
 let homeView = true;
 
+let apiUsers
+let apiRecipes
+let apiIngredients 
+
+
+function fetchData() {
```
Comment (knpausch) [inline src/scripts.js:26]: great promise! lets reorganize this to be with our functions section

File: src/scripts.js [hunk 6]
```diff
@@ -33,12 +55,13 @@ const saveRecipeButton = document.querySelector('#favorite-recipe-button')
 const homeButton = document.querySelector('#home-button')
 const submitButton = document.querySelector('#submit-search-button')
 const searchBar = document.querySelector('#search-bar')
+const deleteInstructions = document.querySelector('#delete-instructions')
 
 
 // ~~~~~~~~~~~~~~ Event Listeners ~~~~~~~~~~~~~~~~~~~~
 allRecipes.addEventListener('click', viewRecipeDetail);
-window.addEventListener('load', displayAllRecipes);
```
Comment (knpausch) [inline src/scripts.js:40]: we can delete these, we no longer use them

**Review by knpausch — APPROVED** [review]
great work on fetch and promise!

File: src/scripts.js [hunk 7]
```diff
@@ -164,18 +187,19 @@ function viewRecipeTotalCost(event) {
     foundRecipe = recipeRepository.recipes.find((current) => {
         return current.id === findId(event);
     })
-    totalCost.innerText = `$ ${foundRecipe.calculateCost(ingredientsData)}`
+    totalCost.innerText = `$ ${foundRecipe.calculateCost(ingredients.ingredients)}`
   };
 
-function randomizeUser() {
-    randomUser = usersData[Math.floor(Math.random() * usersData.length)]
+function randomizeUser(data) {
```
Comment (FrazierLE) [inline src/scripts.js:193]: Is this the api data or the data given?

**Review by FrazierLE — APPROVED** [review]

File: src/classes/Recipe.js (hunk 2 above)
Comment (Eleanorgruth) [reply src/classes/Recipe.js:13]: This is just a parameter!

**Pull request merged**